src/Mantasflowers.WebApi/Controllers/UserController.cs

```diff
@@ -0,0 +1,226 @@
+﻿using FirebaseAdmin.Auth;
+using Mantasflowers.Persistence.Authentication;
+using Mantasflowers.WebApi.Extensions;
+using Mantasflowers.WebApi.Responses;
+using Mantasflowers.WebApi.Setup.Authentication;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Mantasflowers.WebApi.Controllers
+{
+    [ApiController]
+    [Route("/firebase")]
+    public class UserController : ControllerBase
+    {
+        private readonly ILogger _logger = Log.ForContext<UserController>();
+
+        private readonly FirebaseContext _fbContext;
+
+        private readonly IHttpClientFactory _clientFactory;
+
+        private readonly string _requestUri;
+
+        public UserController(FirebaseContext fbContext, IHttpClientFactory clientFactory, IOptionsMonitor<WebApiKey> optionsAccessor)
+        {
+            _fbContext = fbContext;
+            _clientFactory = clientFactory;
+            _requestUri = $"https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key={optionsAccessor.CurrentValue.Value}";
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="email"></param>
+        /// <param name="password"></param>
+        /// <returns></returns>
+        [HttpGet("get-token")]
+        public async Task<IActionResult> GetTokenAsync(string email, string password)
+        {
+            string responseData = string.Empty;
+            try
+            {
+                using var client = _clientFactory.CreateClient();
+
+                string content = $"{{\"email\":\"{email}\",\"password\":\"{password}\",\"returnSecureToken\":true}}";
```

Comment:
I think you can just construct an object (statically typed or anonymous) and pass that to `JsonConvert` instead of doing this manually (less error prone)